client_test.go

```diff
@@ -110,20 +106,26 @@ func TestMain(m *testing.M) {
 	os.Exit(status)
 }
 
-func waitForDaemonStart(ctx context.Context, client *Client) error {
+func waitForDaemonStart(ctx context.Context, address string) (*Client, error) {
 	var (
+		client  *Client
 		serving bool
 		err     error
 	)
 
 	for i := 0; i < 20; i++ {
-		serving, err = client.IsServing(ctx)
-		if serving {
-			return nil
+		if client == nil {
```

Comment:
Has this been moved because `New` will return a `nil` client when it fails to connect?